web/src/app/admin/AdminAPIKeys.tsx

```diff
@@ -0,0 +1,229 @@
+import React, { useState } from 'react'
+import makeStyles from '@mui/styles/makeStyles'
+import {
+  Button,
+  Grid,
+  Typography,
+  Card,
+  ButtonBase,
+  CardHeader,
+} from '@mui/material'
+import { Add } from '@mui/icons-material'
+import AdminAPIKeysDrawer from './admin-api-keys/AdminAPIKeysDrawer'
+import { GQLAPIKey, CreatedGQLAPIKey } from '../../schema'
+import { Time } from '../util/Time'
+import { gql, useQuery } from '@apollo/client'
+import FlatList, { FlatListListItem } from '../lists/FlatList'
+import Spinner from '../loading/components/Spinner'
+import { GenericError } from '../error-pages'
+import { Theme } from '@mui/material/styles'
+import AdminAPIKeysActionDialog from './admin-api-keys/AdminAPIKeysActionDialog'
+import AdminAPIKeysTokenDialog from './admin-api-keys/AdminAPIKeysTokenDialog'
+// query for getting existing API Keys
+const getAPIKeysQuery = gql`
+  query gqlAPIKeysQuery {
+    gqlAPIKeys {
+      id
+      name
+      description
+      createdAt
+      createdBy {
+        id
+        role
+        name
+        email
```

Comment:
We can leave out `role` and `email` that are unused to keep this query smaller

Reply:
Noted and this has been addressed.